src/Monolog/Handler/RotatingFileHandler.php

```diff
@@ -198,4 +198,26 @@ protected function getGlobPattern(): string
 
         return $glob;
     }
+
+    protected function setDateFormat(string $dateFormat): void
+    {
+        if ($dateFormat !== self::FILE_PER_DAY && $dateFormat !== self::FILE_PER_MONTH && $dateFormat !== self::FILE_PER_YEAR) {
+            throw new InvalidArgumentException(
+                'Invalid date format - format must be one of ' .
+                'RotatingFileHandler::FILE_PER_DAY ("Y-m-d"), RotatingFileHandler::FILE_PER_MONTH ("Y-m") ' .
+                'or RotatingFileHandler::FILE_PER_YEAR ("Y"), or you can set one of the ' .
+                'date formats using slashes, underscores and/or dots instead of dashes.'
+            );
+        }
```

Comment:
This validation isn't complete, it should match what is in setFilenameFormat - but I think setFilenameFormat should just call this setDateFormat and then move the validation in here so it is not duplicated.